lifecycle-operator/webhooks/pod_mutator/handlers/pod_annotation.go

```diff
@@ -94,12 +95,18 @@ func copyResourceLabelsIfPresent(sourceResource *metav1.ObjectMeta, targetPod *c
 	postDeploymentChecks, _ = GetLabelOrAnnotation(sourceResource, apicommon.PostDeploymentTaskAnnotation, "")
 	preEvaluationChecks, _ = GetLabelOrAnnotation(sourceResource, apicommon.PreDeploymentEvaluationAnnotation, "")
 	postEvaluationChecks, _ = GetLabelOrAnnotation(sourceResource, apicommon.PostDeploymentEvaluationAnnotation, "")
+	containerName, _ := GetLabelOrAnnotation(sourceResource, apicommon.ContainerNameAnnotation, "")
 
 	if gotWorkloadName {
 		setMapKey(targetPod.Annotations, apicommon.WorkloadAnnotation, workloadName)
 
 		if !gotVersion {
-			setMapKey(targetPod.Annotations, apicommon.VersionAnnotation, calculateVersion(targetPod))
+			version, err := calculateVersion(targetPod, containerName)
+			if err != nil {
```

Comment:
Would it be possible to introduce a test which covers the error case here ?

Reply:
[added test](https://github.com/keptn/lifecycle-toolkit/pull/2471/commits/f72ca17268768005480a58ba9ae437f533ba5946)